server/SupServer.java

```diff
@@ -0,0 +1,86 @@
+import java.io.*;
+import java.net.*;
+import java.util.*;
+
+public class SupServer {
+  
+  // userList is a hashmap which is used to store the user name and its printwriter
+  HashMap<String, PrintWriter> userList = new HashMap<String, PrintWriter>();
```

Comment:
We'll need this map to be thread safe and somewhere all threads can get to it. Could make a [Singleton](http://stackoverflow.com/questions/11165852/java-singleton-and-synchronization) OnlineContacts class with a [ConcurrentHashMap](http://docs.oracle.com/javase/6/docs/api/java/util/concurrent/ConcurrentHashMap.html)?